cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -810,26 +970,72 @@ fn insert_common_epilogue(
     pos: &mut EncCursor,
     reg_type: ir::types::Type,
     csrs: &RegisterSet,
+    isa: &dyn TargetIsa,
+    mut cfa_state: Option<CFAState>,
 ) {
+    let word_size = isa.pointer_bytes() as isize;
     if stack_size > 0 {
         pos.ins().adjust_sp_up_imm(Imm64::new(stack_size));
     }
 
     // Pop all the callee-saved registers, stepping backward each time to
     // preserve the correct order.
     let fp_ret = pos.ins().x86_pop(reg_type);
+    let fp_pop_inst = pos.built_inst();
+
+    if let Some(ref mut cfa_state) = cfa_state.as_mut() {
+        // Account for CFA state in the reverse of `insert_common_prologue`.
+        cfa_state.current_depth += word_size;
+        cfa_state.cf_ptr_offset -= word_size;
+        // And now that we're going to overwrite `rbp`, `rsp` is the only way to get to the call frame.
+        cfa_state.cf_ptr_reg = RU::rsp as RegUnit;
```

Comment:
I feel I've asked this before, but: don't we need to "publish" the changes as FrameLayoutChange here? (and below when popping out nonvolatile regs)

Reply:
We do need to record the layout change here, [and do just that](https://github.com/bytecodealliance/cranelift/pull/1204/files#diff-08ef6b0c3d44ab300b5158dab40ea830R1024-R1027). It's definitely not immediately obvious since tracks frame layout after `pop rbp` but isn't applied until popping other registers.

I [added some comments](https://github.com/bytecodealliance/cranelift/pull/1204/commits/0895a73e9d55b048c9d0fdcf5c57abeaec4b5b63#diff-08ef6b0c3d44ab300b5158dab40ea830R987-L1002) to try clarifying this for future readers.